### ISSUE TITLE
Documentation/request on suppressing warnings for whole types (mostly JS externs)

### ISSUE THREAD

**Issue by vrescobar** [fsprojects/FSharpLint#415]
### Description

Documentation on suppressing warnings for types and structures.

I just read the ticket #343 and for me it is not yet clear how to use the `[<SuppressMessage("*", "RecordFieldNames")>]` code for a whole type structure.

Not sure if this issue falls as documentation issue, feature request or bug.

### Repro steps

I have a project where I use Fable and I have a record which directly maps to an extern javascript object (an answer from an API) which follow the javascript/json naming conventions and I want to make it match my Type directly rather than typing a lot just to keep the linter happy.

If I try to use the `SuppressMessage` it doesn't seem to work (and somehow I couldn't find any documentation):

![Screenshot 2020-05-02 at 07 58 17](https://user-images.githubusercontent.com/6759126/80856482-b166e300-8c4a-11ea-91b6-f085d67e4224.png)

If I want to deactivate the linter only for this object and not the whole file the only thing that works for me is to do it line by line:

### Expected behavior

Some way to deactivate for the following object this sort of warning rather than on line/file basis. Or perhaps inform about how to do it in the documentation.

### Actual behavior
As mentioned.

### Known workarounds
I for sure can do that line by line but by boing that line by line suddenly the code becomes less readable and it is easy to start ignoring meaningful comments due all the information meant to interface the tool. Which I think that defeats the purpose of naming conventions and standardized formatting (making the code readable and using comments only to clarify disambiguations etc).

![Screenshot 2020-05-02 at 08 02 09](https://user-images.githubusercontent.com/6759126/80856544-44a01880-8c4b-11ea-98ea-656253a317c6.png)

### Related information
* VSCode
* Fable project interfacing javascript answers from HTTP
* Compiler mono and webpack
* The documentation I followed is [this one](http://fsprojects.github.io/FSharpLint/Suppression.html)

**Comment by duckmatt** [comment]
Hello, thank you for raising this, it should be possible to wrap the type with comments e.g.

```F#
// fsharplint:disable RecordFieldNames
type MyType =
    { xyz: string
      foo: string }
// fsharplint:enable RecordFieldNames
```

I'll take a look at updating the documentation, the existing sample appears to be broken and the most common case is probably the above so should be covered.

**Comment by vrescobar** [comment]
Thanks for the answer.
I guess that's the smart way, somehow I thought the `disable` was permanently disabling the linter for the rest of the file and didn't realize that the `enable` was actually meant to make such blocks possible. #close
